### PR TITLE
proxy: refactor loadbalanced endpoint selection

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
Rename setRequestURLForLoadBalancedBackend to selectEndpoint and move loadbalancing context creation inside

Followup on #2648

### CODE REVIEW

File: proxy/proxy.go [hunk 1]
```diff
@@ -468,10 +468,18 @@ func setRequestURLForDynamicBackend(u *url.URL, stateBag map[string]interface{})
 	}
 }
 
-func setRequestURLForLoadBalancedBackend(u *url.URL, rt *routing.Route, lbctx *routing.LBContext) *routing.LBEndpoint {
+func selectEndpoint(ctx *context, registry *routing.EndpointRegistry) *routing.LBEndpoint {
+	rt := ctx.route
+
```
Comment (AlexanderYastrebov) [inline proxy/proxy.go:473]: Here we can put fadeIn and passive healthchecks

File: proxy/proxy.go [hunk 2]
```diff
@@ -468,10 +468,18 @@ func setRequestURLForDynamicBackend(u *url.URL, stateBag map[string]interface{})
 	}
 }
 
-func setRequestURLForLoadBalancedBackend(u *url.URL, rt *routing.Route, lbctx *routing.LBContext) *routing.LBEndpoint {
+func selectEndpoint(ctx *context, registry *routing.EndpointRegistry) *routing.LBEndpoint {
+	rt := ctx.route
```
Comment (RomanZavodskikh) [inline proxy/proxy.go:472]: minor: I don't understand why do we need rt variable here and why cannot we inline `ctx.route`?

File: proxy/proxy.go [hunk 3]
```diff
@@ -490,7 +498,9 @@ func mapRequest(ctx *context, requestContext stdlibcontext.Context, removeHopHea
 		setRequestURLFromRequest(u, r)
 		setRequestURLForDynamicBackend(u, stateBag)
 	case eskip.LBBackend:
-		endpoint = setRequestURLForLoadBalancedBackend(u, rt, &routing.LBContext{Request: r, Route: rt, LBEndpoints: rt.LBEndpoints, Params: stateBag, Registry: registry})
+		endpoint = selectEndpoint(ctx, registry)
+		u.Scheme = endpoint.Scheme
```
Comment (RomanZavodskikh) [inline proxy/proxy.go:502]: minor: maybe use fallthrough here?

**Comment by RomanZavodskikh** [comment]
:+1:

File: proxy/proxy.go (hunk 2 above)
Comment (AlexanderYastrebov) [reply proxy/proxy.go:472]: rt.Apply

File: proxy/proxy.go (hunk 3 above)
Comment (AlexanderYastrebov) [reply proxy/proxy.go:502]: no need to overcomplicate to save 1 line

Comment (AlexanderYastrebov) [reply proxy/proxy.go:502]: also it uses endpoint fields, not route fields

**Comment by AlexanderYastrebov** [comment]
:+1:

**Pull request merged**